app/queries/find_deep_errored_file_sets.rb

```diff
@@ -12,7 +12,7 @@ def initialize(query_service:)
   end
 
   def find_deep_errored_file_sets(resource:)
-    query_service.connection[relationship_query, id: resource.id.to_s].first[:count]
```

Comment:
Suggestion: Maybe we should have a `count: true` option? I think this changes the query to always realize every errored file set as a real resource, which is more ruby-expensive than just returning the count. I realize you also need the resources for this case, though, so maybe a param to switch between those states.

Reply:
That's a good point. Pushed up a commit to address this.